src/Nethermind/Nethermind.Synchronization/SnapSync/SnapSyncFeed.cs

```diff
@@ -205,6 +205,7 @@ public override void SyncModeSelectorOnChanged(SyncMode current)
             if (_disposed) return;
             if (CurrentState == SyncFeedState.Dormant)
             {
+                _snapProvider.Dispose();
```

Comment:
won't this be called over and over again?

Reply:
Yes.

Reply:
Ow, I think I made a mistake here.